itest/starboard-operator/starboard_operator_test.go

```diff
@@ -15,55 +15,80 @@ import (
 	"k8s.io/utils/pointer"
 )
 
+const (
+	assertionTimeout = 3 * time.Minute
+)
+
 var _ = Describe("Starboard Operator", func() {
 
 	const (
 		namespaceName  = corev1.NamespaceDefault
 		deploymentName = "wordpress"
 	)
 
-	It("Should create VulnerabilityReport when Deployment is created", func() {
-		_, err := kubernetesClientset.AppsV1().Deployments(namespaceName).
-			Create(context.TODO(), &appsv1.Deployment{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      deploymentName,
-					Namespace: namespaceName,
-				},
-				Spec: appsv1.DeploymentSpec{
-					Replicas: pointer.Int32Ptr(1),
-					Selector: &metav1.LabelSelector{
-						MatchLabels: map[string]string{"app": "wordpress"},
+	Describe("When a new Deployment is created", func() {
+
+		BeforeEach(func() {
+			_, err := kubeClientset.AppsV1().Deployments(namespaceName).
+				Create(context.TODO(), &appsv1.Deployment{
```

Comment:
nit: What if you define `ctx` at first and use it for each method? `context.TODO` makes me feel like you are planning to pass the upstream `ctx` in the future.

```suggestion
	Describe("When a new Deployment is created", func() {
		ctx := context.Background()
		BeforeEach(func() {
			_, err := kubeClientset.AppsV1().Deployments(namespaceName).
				Create(ctx, &appsv1.Deployment{
```

>TODO returns a non-nil, empty Context. Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

https://golang.org/pkg/context/#TODO

In this case, it doesn't seem that the same `ctx` needs to be shared, so you can just pass `context.Background()`.

```suggestion
	Describe("When a new Deployment is created", func() {

		BeforeEach(func() {
			_, err := kubeClientset.AppsV1().Deployments(namespaceName).
				Create(context.Background(), &appsv1.Deployment{
```

Reply:
That's a good point. Agreed.